### PR TITLE
Hidden lairs

### PR DESCRIPTION
**Pull request by OH296**
- complete restructure of p_upgrades e.g arsenals lairs and gene vaults making them structs
- redone saving structure for p_upgrades
- most importantly makes them stable and solves all game crashes however there are still some visual tweaks that i'll get onto

### CODE REVIEW

File: scripts/scr_planetary_feature/scr_planetary_feature.gml [hunk 1]
```diff
@@ -48,8 +48,27 @@ function new_planet_feature(feature_type) constructor{
...
 		planet_display = "Hidden Secret Base";
-		player_hidden = 0
+		player_hidden = 0;
+		style = "UTL"
+		if (struct_exists(other_data, "style")){
+			style = other_data[$ "style"];
+		}
+		built = obj_controller.turn +3;
+		forge = 0;hippo=0;beastarium=0;torture=0;narcotics=0;relic=0;cookery=0;
+		vox=0;librarium=0;throne=0;stasis=0;swimming=0;stock=0;
```
[the remark this answers is not in the document]
Comment (OH296) [reply scripts/scr_planetary_feature/scr_planetary_feature.gml:58]: gotcha

File: scripts/scr_ui_popup/scr_ui_popup.gml [hunk 2]
```diff
@@ -14,8 +14,14 @@ function scr_ui_popup() {
     
 	    draw_set_font(fnt_40k_30b);
 	    draw_set_halign(fa_center);
-	    var title,ups;title="";ups=obj_temp_build.lair+obj_temp_build.arsenal+obj_temp_build.gene_vault;
-    
+	    var title,un_upgraded;title="";
+	    var planet_upgrades = obj_temp_build.target.p_upgrades[obj_controller.selecting_planet];
+	    un_upgraded = 0;
```
[the remark this answers is not in the document]
Comment (OH296) [reply scripts/scr_ui_popup/scr_ui_popup.gml:19]: done

File: scripts/scr_ui_popup/scr_ui_popup.gml [hunk 3]
```diff
@@ -14,8 +14,14 @@ function scr_ui_popup() {
...
+	    var arsenal = 0; gene_vault=0;s_base=0;
```
[the remark this answers is not in the document]
Comment (OH296) [reply scripts/scr_ui_popup/scr_ui_popup.gml:20]: done

File: scripts/scr_ui_popup/scr_ui_popup.gml [hunk 4]
```diff
@@ -45,208 +51,207 @@ function scr_ui_popup() {
...
+			            if (yep=false) then woob+="a massive throne, though it is currently vacant.  ";
+			        }
+			        if (s_base.vox>0) and (fuck.target.p_player[obj_controller.selecting_planet]>0) then woob+="Heretical music blasts from the vox-casters, shaking the walls.  ";
+			        if (s_base.narcotics>0) then woob+="  Many of the tables have lines of white powder set on paper or bunches of needles.  Plastic straws lay close by.  ";
+			        if (s_base.cookery=1){
+			            if (fuck.target.p_player[obj_controller.selecting_planet]>0) then woob+="Imperial Chefs are currently bustling to and from the kitchen, cooking savory treats and food for those present.  ";
+			            if (fuck.target.p_player[obj_controller.selecting_planet]=0) then woob+="The Imperial Chefs are mostly idle, making use of the other rooms and facilities.  ";
+			        }
+		        
+			        if (s_base.stock=1) then woob+="  One of the chambers is hollowed out to display war trophies and gear.  ";
```
[the remark this answers is not in the document]
Comment (OH296) [reply scripts/scr_ui_popup/scr_ui_popup.gml:173]: used else if because of conditional arguments

**Comment by OH296** [comment]
alright should have hit most all of those changes

**Review by GalacticChimp — APPROVED** [review]

**Pull request merged**